core/desugarEffects.ml

```diff
@@ -978,6 +982,7 @@ class main_traversal simple_tycon_env =
       let module D = Datatype in
       let o, rv =
         match rv with
+        | D.EffectApplication _ -> (o, rv) (* TODO(rj) do i need to do something there ? *)
```

Comment:
You need to make sure the alias constructor is desugared and then make sure the arguments are desugared.